File: leetcode/2100-2199/2109. Adding Spaces to a String/2109.cs
public class Solution
{
    public string AddSpaces(string s, int[] spaces)
    {
        int i = 0, j = 0, k = 0;
        int n = s.Length, m = spaces.Length;
        var res = new char[n + m];
        while (i < n)
        {
            if (j < m && i == spaces[j])
            {
                res[k++] = ' ';
                j++;
            }
            res[k++] = s[i++];
        }

        return new String(res);
    }
}

// TC:O(n+m) && SC:O(n+m)

File: leetcode/2100-2199/2109. Adding Spaces to a String/2109.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        res = []
        j = 0
        for i, char in enumerate(s):
            if j < len(spaces) and i == spaces[j]:
                res.append(" ")
                j += 1

            res.append(char)

        return "".join(res)

# TC:O(n+m) and SC:O(n+m)